src/common/containers/ChartContainer.tsx

```diff
@@ -175,6 +181,12 @@ export const ChartContainer: FC<ChartContainerProps> = ({
   const styleWidth =
     rest.width !== undefined && rest.width !== null ? rest.width : '100%';
 
+  useEffect(() => {
+    if (onChartDimensionsChanged) {
+      onChartDimensionsChanged({ chartHeight: childProps.chartHeight, chartWidth: childProps.chartWidth });
+    }
+  }, [childProps.chartHeight, childProps.chartWidth, onChartDimensionsChanged]);
+
```

Comment:
This is the most important piece of the change. It's important to put this callback function which triggers the parent state change inside a useEffect so that it's not triggered during the render phase (which causes the warning)